### ISSUE TITLE
Version 2.8.0 broke pip source installation

### ISSUE THREAD

**Issue by stephan-cr** [pydata/numexpr#392]
Whenever numexpr is installed from source, installation breaks with the following error message:

```shell
$ python -m pip install numexpr==2.8.0 --no-binary :all:
Collecting numexpr==2.8.0
  Using cached numexpr-2.8.0.tar.gz (95 kB)
  Preparing metadata (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /home/stephan/venvs/test/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-sqeas1oi/numexpr_c74523878e7b43239a23676f9e5e1a5c/setup.py'"'"'; __file__='"'"'/tmp/pip-install-sqeas1oi/numexpr_c74523878e7b43239a23676f9e5e1a5c/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-suc2ri72
       cwd: /tmp/pip-install-sqeas1oi/numexpr_c74523878e7b43239a23676f9e5e1a5c/
  Complete output (5 lines):
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "/tmp/pip-install-sqeas1oi/numexpr_c74523878e7b43239a23676f9e5e1a5c/setup.py", line 16, in <module>
      import numpy as np
  ModuleNotFoundError: No module named 'numpy'
  ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/6f/18/4f07b0f5effb3b6fe4611738c15ce208f36dfe140c1f3ba1f28e62294717/numexpr-2.8.0.tar.gz#sha256=9fec076b76c90a5f3929373f548834bb203c6d23a81a895e60d0fe9cca075e99 (from https://pypi.org/simple/numexpr/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
ERROR: Could not find a version that satisfies the requirement numexpr==2.8.0 (from versions: 1.1.1, 1.3, 1.3.1, 1.4, 1.4.1, 1.4.2, 2.0.1, 2.1, 2.2.1, 2.2.2, 2.3rc1, 2.3, 2.3.1, 2.4rc1, 2.4rc2, 2.4, 2.4.1, 2.4.2, 2.4.3, 2.4.4, 2.4.5, 2.4.6, 2.5, 2.5.1, 2.5.2, 2.6.0, 2.6.1, 2.6.2, 2.6.3, 2.6.4, 2.6.5, 2.6.6.dev0, 2.6.6, 2.6.7, 2.6.8, 2.6.9, 2.7.0, 2.7.1, 2.7.2, 2.7.3, 2.8.0)
ERROR: No matching distribution found for numexpr==2.8.0
```

The previous version 2.7.3 works just fine:

```shell
$ python -m pip install numexpr==2.7.3 --no-binary :all:
Collecting numexpr==2.7.3
  Using cached numexpr-2.7.3.tar.gz (97 kB)
  Preparing metadata (setup.py) ... done
Collecting numpy>=1.7
  Using cached numpy-1.21.4.zip (10.6 MB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Skipping wheel build for numexpr, due to binaries being disabled for it.
Building wheels for collected packages: numpy
  Building wheel for numpy (pyproject.toml) ... done
  Created wheel for numpy: filename=numpy-1.21.4-cp39-cp39-linux_x86_64.whl size=17290601 sha256=614bc3d4c992ab1933a2a7bc4a08be6502030098ed68842a629da694489d6165
  Stored in directory: /home/stephan/.cache/pip/wheels/14/97/c6/165e4deb0c59cfe0601822a5352d86f3ec79285d6429a07c1c
Successfully built numpy
Installing collected packages: numpy, numexpr
    Running setup.py install for numexpr ... done
Successfully installed numexpr-2.7.3 numpy-1.21.4
```

The installation via wheel is not affected as well:

```shell
$ python -m pip install numexpr
Collecting numexpr
  Downloading numexpr-2.8.0-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (380 kB)
     |████████████████████████████████| 380 kB 2.0 MB/s
Collecting numpy>=1.7
  Downloading numpy-1.21.4-cp39-cp39-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (15.7 MB)
     |████████████████████████████████| 15.7 MB 2.2 MB/s
Installing collected packages: numpy, numexpr
Successfully installed numexpr-2.8.0 numpy-1.21.4
```

**Comment by stephan-cr** [comment]
Pointed out by @emjohnson20 in #391 as well.

**Comment by stephan-cr** [comment]
I think the package needs a pyproject.toml, which roughly contains:

```toml
[build-system]
requires = ["setuptools", "wheel", "numpy"]
```

Some version restrictions might be required as well.

**Comment by robbmcleod** [comment]
I've managed to get MKL installed on my laptop so I'll see about testing and pushing a 2.8.1 release this week, if time permits.

**Comment by robbmcleod** [comment]
Well the description on PyPi was somehow mangled, not sure what's up with that but it's not a serious issue. The `setup.cfg` was setup for RST when I first uploaded the sdist and then it didn't seem to update correctly when I uploaded it again. Otherwise closing this issue unless we have more problems with 2.8.1 release.

**Comment by avalentino** [comment]
> Well the description on PyPi was somehow mangled, not sure what's up with that but it's not a serious issue. The setup.cfg was setup for RST when I first uploaded the sdist and then it didn't seem to update correctly when I uploaded it again. Otherwise closing this issue unless we have more problems with 2.8.1 release.

@robbmcleod sorry my fault.
I have used the incorrect value for the `long_description_content_type` in `setup.cfg`.
It should be fixed in #394.
